lib/mauth/request_and_response.rb

```diff
@@ -35,13 +112,27 @@ def attributes_for_signing
   # methods for an incoming object which is expected to have a signature.
   #
   # includer must provide
+  # - #mcc_authentication which returns that header's value
+  # - #mcc_time
+  # OR
   # - #x_mws_authentication which returns that header's value
   # - #x_mws_time
   module Signed
-    # returns a hash with keys :token, :app_uuid, and :signature parsed from the X-MWS-Authentication header
+    # mauth_client will authenticate with the highest protocol version present and ignore other
+    # protocol versions.
+    # returns a hash with keys :token, :app_uuid, and :signature parsed from the MCC-Authentication header
+    # if it is present and if not then the X-MWS-Authentication header if it is present.
+    # Note MWSV2 protocol no longer allows more than one space between the token and app uuid.
     def signature_info
       @signature_info ||= begin
-        match = x_mws_authentication && x_mws_authentication.match(/\A([^ ]+) *([^:]+):([^:]+)\z/)
+        match = if mcc_authentication
+          mcc_authentication.match(
+            /\A(#{MAuth::Client::MWSV2_TOKEN}) ([^:]+):([^:]+)#{MAuth::Client::AUTH_HEADER_DELIMITER}\z/
```

Comment:
If we specify the length of the app uuid, signature, or whole header then we would be throwing out invalid signatures and app uuids here before we get to the authenticator validations which could make logs and errors less helpful.
One example: if an app_uuid is too short we could end up getting 
`"mAuth service responded with 404 looking up public key for "` rather than 
`"mAuth service responded with 404 looking up public key for 123"`